lib/RewriteInsertsPass.cpp

```diff
@@ -323,7 +371,9 @@ bool RewriteInsertsPass::ReplacePartialInsertions(Module &M) {
     // Record the list of tails for the next round.
     InsertionVector NextRoundWorkList;
 
-    for (InsertValueInst *insertion : WorkList) {
+    //for (InsertValueInst *insertion : WorkList) {
+    for (Instruction *inst : WorkList) {
+      InsertValueInst *insertion = cast<InsertValueInst>(inst);
```

Comment:
This cast will assert out if it's incorrect.  So why not keep the old more-specific type declaration in the range-for?